Tests/AnyCodableTests/AnyEncodableTests.swift

```diff
@@ -66,9 +66,47 @@ class AnyEncodableTests: XCTestCase {
         XCTAssert(encodedJSONObject["integer"] is Int)
         XCTAssert(encodedJSONObject["double"] is Double)
     }
+    
+    #if swift(>=5.0)
+    func testStringInterpolationEncoding() throws {
+        let dictionary: [String: AnyEncodable] = [
+            "boolean": "\(true)",
+            "integer": "\(1)",
+            "double": "\(3.141592653589793)",
```

Comment:
not sure if the precision of a string interpolation of a float is defined,
this is what worked on macOS with Swift 5.2

Reply:
BTW the existing test `AnyEncodableTests.testJSONEncoding` passes on macOS+Swift5.2 (and 5.3beta) but fails on:
```
Swift version 5.2.4 (swift-5.2.4-RELEASE)
Target: x86_64-unknown-linux-gnu
```

```
/src/Tests/AnyCodableTests/AnyEncodableTests.swift:40: error: AnyEncodableTests.testJSONEncoding : XCTAssertEqual failed: ("{
    nested =     {
        a = alpha;
        c = charlie;
        b = bravo;
    };
    integer = 1;
    string = string;
    double = 3.141592653589793;
    boolean = true;
    array =     (
        1,
        2,
        3
    );
}") is not equal to ("{
    double = 3.14159265358979323846;
    array =     (
        1,
        2,
        3
    );
    integer = 1;
    string = string;
    boolean = true;
    nested =     {
        a = alpha;
        c = charlie;
        b = bravo;
    };
}") -
Test Case 'AnyEncodableTests.testJSONEncoding' failed (0.001 seconds)
```

Reply:
The reason for this failure has to do with platform differences in JSON decoding for floating-point numbers; notice the expected value `3.141592653589793` is different from the actual `3.14159265358979323846`. To work around this, you could use [`XCTAssertEqual(_:_:accuracy:_:file:line:)`](https://developer.apple.com/documentation/xctest/2919914-xctassertequal) with a delta >= 1e-7.

Reply:
the accuracy cannot be easily used when comparing strings, right?

Reply:
Yes. And to correct my previous statement, I got it wrong — this is a platform difference of how floating-point numbers are _encoded_, not decoded. The way to fix this is to use a floating point number that won't have the same precision errors. For example, `1.2`.